src/Options/default_options.jl

```diff
@@ -92,7 +92,7 @@ function validate_solver_options_and_add_default_values!(options::Options)::Opti
                 error("No property has been defined.")
             end
         elseif key == :property
-            expected_type = Union{Nothing, Property, Dict{Int, <:Property}}
+            expected_type = Union{Nothing, Property, Dict{Int, <:Property}, Function}
```

Comment:
Here we may add a new concrete subtype that wraps a function.

Reply:
Here are three options. I tend toward 3.
1. Add `SafeFunctionProperty`.
2. Generalize the existing `SafeStatesProperty` to allow also storing a function instead of a set. We would have to add a field for the dimension `n`.
3. Merge `SafeStatesProperty` and `BadStatesProperty` into `LazySetProperty` and add a flag expressing whether the set represents the safe states or the bad states.

Reply:
Follow-up: https://github.com/JuliaReach/Reachability.jl/issues/566